### PR TITLE
i2p: 1.9.0 -> 2.0.0

### PR DESCRIPTION
**Pull request by gp2112**
###### Description of changes

- i2ptunnel: Support SHA-256 digest proxy authentication (RFC 7616)
- SSU2: Connection migration
- SSU2: Immediate acks
- SSU2: Enable by default

Full release description: https://geti2p.net/en/blog/post/2022/11/21/2.0.0-Release

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by dasj19 — APPROVED** [review]
Built and ran on x86_64

**Comment by gp2112** [comment]
ping @joelmo ?
I'll add myself as a maintainer, if that's ok

**Comment by linsui** [comment]
Please remove the unrelated package.

**Comment by linsui** [comment]
The I2P update works well for me.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
As already mentioned feather must be removed. The other commits can be combined.

File: pkgs/applications/blockchains/feather/default.nix [hunk 1]
```diff
@@ -0,0 +1,99 @@
+{ lib, stdenv, cmake, ccache
```
Comment (dasj19) [inline pkgs/applications/blockchains/feather/default.nix:1]: Please make another for this one, here we focus on only on i2p.

**Pull request closed without merging**